packages/langium/src/workspace/environment.ts

```diff
@@ -0,0 +1,51 @@
+/******************************************************************************
+ * Copyright 2024 TypeFox GmbH
+ * This program and the accompanying materials are made available under the
+ * terms of the MIT License, which is available in the project root.
+ ******************************************************************************/
+
+import type { InitializeParams, InitializedParams } from 'vscode-languageserver-protocol';
+
+export interface EnvironmentInfo {
+    readonly isLanguageServer: boolean;
+    readonly locale: string;
+}
+
+export interface Environment extends EnvironmentInfo {
```

Comment:
Do we really need this as a separate service? AFAICT the information is only used by the WorkspaceManager.

Reply:
I feel like this could benefit other services as well and is well suited as a stand-alone service.